Objects/genobject.c

```diff
@@ -814,8 +813,7 @@ static PyAsyncMethods gen_as_async = {
 PyTypeObject PyGen_Type = {
     PyVarObject_HEAD_INIT(&PyType_Type, 0)
     "generator",                                /* tp_name */
-    offsetof(PyGenObject, gi_iframe) +
-    offsetof(_PyInterpreterFrame, localsplus),       /* tp_basicsize */
+    sizeof(PyGenObject),                        /* tp_basicsize */
```

Comment:
I think this should be `offsetof(PyGenObject, gi_frame.localsplus)` otherwise we are overallocating one `PyObject *` slot.

Likewise for `PyCoroObject` and `PyAsyncGenObject`

Reply:
Let's try that: https://github.com/python/cpython/pull/120941